### ISSUE TITLE
HttpProxy server side Criteria object deserialization

### ISSUE THREAD

**Issue by MarceloJGaray** [MarimerLLC/cslaforum#280]
I'm currently trying to switch an existing WPF application from WCFProxy to HttpProxy. I did the configuration on the MVC5 server as per this thread https://github.com/MarimerLLC/cslaforum/issues/19, and also following the SimpleApp sample, but I'm having a problem with the Criteria object deserialization on the server side. DataPortal_Fetch calls the parameterless Criteria constructor instead of the one with parameters (username, password) like the WCFProxy does. I'm using custom principal and custom identity that works just fine with the WCFProxy. I've put a breakpoint on the PostAsync override on DataPortalController on the server side and seems to be called just fine.

I'm pretty sure I'm missing something on the HttpProxy configuration but not sure what it is. Any ideas?

**Comment by rockfordlhotka** [comment]
Deserialization is a special form of creating an object, and different serializers handle this differently: but they all avoid the use of "standard" constructors with parameters.

BinaryFormatter and NDCS use a special serializer that allows them to pass in a serialization info object. MobileFormatter (likely what you are using) uses the default (parameterless) constructor. In all three cases the serializer then uses expression trees, reflection, or other advanced techniques to directly load the fields of the new object instance with the values from the serialized byte stream.

In the case of MobileFormatter, if you are using managed backing fields then things should be automatic: the serializer will automatically load your properties with the values from the serialization stream. If you are using private backing fields then you must manually overload methods to get/set the state of your fields in the serialization stream. This is discussed in the _Using CSLA 4_ ebooks.

**Comment by MarceloJGaray** [comment]
Rocky,

Thanks for the clarification. I got that much from Using CSLA 4 ebooks and the old forum, but I was confused by the fact that some of my business objects didn’t get that error and some did (all with managed backing fields only). Then I realized that some CSLA-based classes (like that ones that inherit from CommandBase) that already have a constructor with parameters also NEED the public parameterless constructor in order to avoid that error. All good now.

BTW: I was at VS Live Orlando last week. It was a great conference, especially the Modern Apps track. Congrats!
